process/jq.go

```diff
@@ -19,13 +20,35 @@ const errJqNoOutputGenerated = errors.Error("no output generated")
 type procJQ struct {
 	process
 	Options procJQOptions `json:"options"`
+
+	query *gojq.Query
 }
 
 type procJQOptions struct {
 	// Query is the jq query applied to data.
 	Query string `json:"query"`
 }
 
+// Create a new join processor.
+func newProcJQ(cfg config.Config) (p procJQ, err error) {
+	err = config.Decode(cfg.Settings, &p)
+	if err != nil {
+		return procJQ{}, err
+	}
+
+	p.operator, err = condition.NewOperator(p.Condition)
+	if err != nil {
+		return procJQ{}, err
+	}
+
+	p.query, err = gojq.Parse(p.Options.Query)
+	if err != nil {
+		return procJQ{}, err
```

Comment:
refactor: required option error

Reply:
I'll check for an empty query/missing option above and then handle the parsing error the same since that could also be from an invalid JQ query.